### ISSUE TITLE
Regression: spacing rules for functions

### ISSUE THREAD

**Issue by pp-gborodin** [sqlfluff/sqlfluff#6428]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have this spark sql query 
```sql
select /*+ repartition(200) */
    one,
    two
from
    mytable
```
and starting from commit 18293ce419e9605e978e47b7f3e83fd7543f4818 sqlfluff is failing with
```
== [query.sql] FAIL                                                                                                                                                                                    
L:   1 | P:  23 | LT01 | Expected single whitespace between function name
                       | identifier and start bracket '('.
                       | [layout.spacing]
```

### Expected Behaviour

I expect sqlfluff to ignore spacing in the query comments. Or there should be an option to disable this behavior

### Observed Behaviour

sqlfluff is failing with
```
== [query.sql] FAIL                                                                                                                                                                                    
L:   1 | P:  23 | LT01 | Expected single whitespace between function name
                       | identifier and start bracket '('.
                       | [layout.spacing]
```

### How to reproduce

Create file `query.sql`:
```sql
select /*+ repartition(200) */
    one,
    two
from
    mytable
```
Then execute
```
git checkout 18293ce419e9605e978e47b7f3e83fd7543f4818
uv venv
uv pip install .
uv run sqlfluff lint --dialect sparksql ./query.sql 
```

### Dialect

sparksql

### Version

version 3.1.1 didn't have the issue, version 3.2.0 is the first release that has the issue
I bisected the problem to commit 18293ce419e9605e978e47b7f3e83fd7543f4818 

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pp-gborodin** [comment]
cc @WittierDinosaur

**Comment by pp-gborodin** [comment]
Thank you for the fix! :heart:
